src/pai_rag/evaluation/generator/rag_qca_sample.py

```diff
@@ -0,0 +1,103 @@
+    )
+    reference_contexts: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The contexts used to generate the reference answer.",
+    )
+    reference_node_id: Optional[List[str]] = Field(
+        default_factory=None, description="The node id corresponding to the contexts"
+    )
+    reference_answer: str = Field(
+        default_factory=str,
+        description="The reference (ground-truth) answer to the example.",
+    )
+    reference_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the reference answer."
+    )
+
+    @property
+    def class_name(self) -> str:
+        """Data example class name."""
+        return "RagQCADataset"
+
+
+class PredictedRagQCASample(BaseLlamaDataExample):
+    """RAG example class. Analogous to traditional ML datasets, this dataset contains
+    the "features" (i.e., query + context) to make a prediction and the "label" (i.e., response)
+    to evaluate the prediction.
+
+    Args:
+        query (str): The user query
+        query_by (CreatedBy): Query generated by human or ai (model-name)
+        reference_contexts (Optional[List[str]]): The contexts used for response
+        reference_node_id (Optional[List[str]]): The node id corresponding to the contexts
+        reference_answer ([str]): Reference answer to the query. An answer
+                                    that would receive full marks upon evaluation.
+        reference_answer_by: The reference answer generated by human or ai (model-name).
+    """
+
+    query: str = Field(
+        default_factory=str, description="The user query for the example."
+    )
+    query_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the query."
+    )
+    reference_contexts: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The contexts used to generate the reference answer.",
+    )
+    reference_node_id: Optional[List[str]] = Field(
+        default_factory=None, description="The node id corresponding to the contexts"
+    )
+    reference_answer: str = Field(
+        default_factory=str,
+        description="The reference (ground-truth) answer to the example.",
+    )
+    reference_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the reference answer."
+    )
+    predicted_contexts: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The contexts used to generate the predicted answer.",
+    )
+    predicted_node_id: Optional[List[str]] = Field(
+        default_factory=None,
+        description="The node id corresponding to the predicted contexts",
+    )
+    predicted_answer: str = Field(
+        default_factory=str,
+        description="The predicted (ground-truth) answer to the example.",
+    )
+    predicted_answer_by: Optional[CreatedBy] = Field(
+        default=None, description="What generated the predicted answer."
```

Comment:
Maybe "who generated the predicted answer?"

Reply:
Fixed.